framework/db/QueryBuilder.php

```diff
@@ -278,6 +278,9 @@ public function batchInsert($table, $columns, $rows)
                 }
                 if (is_string($value)) {
                     $value = $schema->quoteValue($value);
+                } elseif (is_float($value)) {
+                    // ensure type cast always has . as decimal separator in all locales
+                    $value = str_replace(',', '.', (string) $value);
```

Comment:
pretty sure :)

```php
<?php

// Usage: install all locales: https://packages.debian.org/stretch/locales-all
// apt-get install locales-all
// then run the script

$locales = `localectl list-locales`;
$locales = array_filter(explode("\n", $locales));

//print_r($locales);
//
foreach($locales as $locale) {
	setlocale(LC_NUMERIC, $locale);
	echo (string) 1234.5678;
	// echo " $locale ";
	echo "\n";
}
```
```
$ php locales.php | sort | uniq
1234,5678
1234.5678
1234�5678
```

`1234�5678` refers to [ps](https://en.wikipedia.org/wiki/Pashto)_[AF](https://en.wikipedia.org/wiki/Afghanistan) which looks like a bug in their locale data because the sign is the UTF-8 replacement character. https://unicode-table.com/en/FFFD/